File: crates/corelink-wasm/examples/quickstart_portal_session.ts
// quickstart_portal_session — Bootstrap an enterprise-portal session
// via POST /v1/enterprise/inquire.
//
// Customer concept: enterprise prospects request contracted-tier
// provisioning (DPA + BAA + custom SLA). Returns a portal URL +
// short-lived session token.
//
// Run: CORELINK_PAT=$PAT tsx examples/quickstart_portal_session.ts

const api = process.env.CORELINK_API_URL ?? "https://sandbox.corelink.humangr.com";
const pat = process.env.CORELINK_PAT;
if (!pat) {
  console.error("error: CORELINK_PAT env var is required");
  process.exit(2);
}
const idem = `idem-${crypto.randomUUID()}`;

try {
  const resp = await fetch(`${api}/v1/enterprise/inquire`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${pat}`,
      "Content-Type": "application/json",
      "Idempotency-Key": idem,
    },
    body: JSON.stringify({
      company: "Acme Corp",
      contact_email_hash: "0".repeat(64),
      expected_seats: 250,
      use_case: "monorepo build cache for 80 engineers",
    }),
  });
  const text = await resp.text();
  console.log(`status=${resp.status} body=${text}`);
} catch (err) {
  console.error(`network_error: ${err instanceof Error ? err.message : String(err)}`);
  process.exit(1);
}

File: tools/cli/examples/quickstart_portal_session.rs
//! `quickstart_portal_session` — Bootstrap an enterprise-portal session via
//! `POST /v1/enterprise/inquire`.
//!
//! Customer concept: enterprise prospects use the portal to request
//! contracted-tier provisioning (DPA + BAA + custom SLA). This call
//! returns a portal URL + short-lived session token; the prospect
//! completes the questionnaire in-browser.
//!
//! Run:
//!
//! ```text
//! CORELINK_PAT=$PAT \
//!   cargo run --example quickstart_portal_session -p corelink-cli
//! ```

#![forbid(unsafe_code)]
#![allow(
    clippy::print_stdout,
    reason = "examples emit human-readable narration to stdout"
)]

use anyhow::{Context, Result};
use serde_json::json;
use std::env;
use uuid::Uuid;

#[tokio::main]
async fn main() -> Result<()> {
    let api = env::var("CORELINK_API_URL")
        .unwrap_or_else(|_| "https://sandbox.corelink.humangr.com".to_string());
    let pat = env::var("CORELINK_PAT").context("CORELINK_PAT is required")?;
    let idem = format!("idem-{}", Uuid::now_v7());

    let body = json!({
        "company": "Acme Corp",
        "contact_email_hash": "0".repeat(64),
        "expected_seats": 250,
        "use_case": "monorepo build cache for 80 engineers",
    });

    let resp = reqwest::Client::new()
        .post(format!("{api}/v1/enterprise/inquire"))
        .bearer_auth(&pat)
        .header("Idempotency-Key", &idem)
        .json(&body)
        .send()
        .await
        .context("POST /v1/enterprise/inquire")?;

    let status = resp.status();
    let text = resp.text().await.context("read response body")?;
    println!("status={status} body={text}");
    Ok(())
}
